### DISCUSSION TITLE
Custom components and conditions

### DISCUSSION THREAD

**Discussion by filippoferrari** [mwshinn/PyDDM#83] [Ask for help] [answered]
Hi, 

I am trying to fit a simple DDM (constant drift, noise, bound and non-decision time) on some experimental data consisting of two types of trials - we expect to see some differences in some of the parameters across these two conditions (for example non-decision time differences between the two types of trials). 
In the examples, I've seen that to achieve this you are required to define your own model components (like DriftCoherence or DriftCoherenceLeak). 
I was wondering what would be the correct way to define the most "general" custom component (i.e., I'm just interested in a difference across these two conditions; since, based on the examples, having a linear dependency between the differences and the trial type wouldn't make much sense in our case). 
Is there anything similar to the `depends_on={'v': 'stim'}` flag in HDDM or an identity design matrix in DMAT?
I could fit the two types of trials independently, but I would like to have the same fitted parameters across conditions for those we don't expect to differ. 

Thanks!
Filippo

**Comment by mwshinn** [comment, marked as answer]
Hi Filippo,

How about something like this?

```
from pyddm import Drift
class DriftCondition(Drift):
    name = "Drift depends on a condition value"
    required_parameters = ["drift1", "drift2", "drift3"]
    required_conditions = ["cond"] # <--Can be 1, 2, or 3
    def get_drift(self, conditions, **kwargs):
        if conditions['cond'] == 1:
            return self.drift1
        elif conditions['cond'] == 2:
            return self.drift2
        elif conditions['cond'] == 3:
            return self.drift3
```

**Reply by filippoferrari** [reply]
That's perfect, thank you!
